### PR TITLE
Add missing generated small buttons and adjust some others

### PR DESCRIPTION
**Pull request by zenseii**
Add UNIFORM exit button for market place, add okay button for necromancy dialog at end of battle. Fix size of many buttons to fit smallest possible words for French.

### CODE REVIEW

**Comment by ihhub** [comment]
Hi @zenseii , how important this pull request to be in 1.0?

File: src/fheroes2/agg/agg_image.cpp [hunk 1]
```diff
@@ -818,8 +822,33 @@ namespace fheroes2
                     break;
                 }
 
-                int32_t textWidth = 87;
-                createNormalButton( _icnVsSprite[id][0], _icnVsSprite[id][1], textWidth, gettext_noop( "TRADE" ), isEvilInterface );
+                const int baseIcnID = isEvilInterface ? ICN::SYSTEME : ICN::SYSTEM;
+
+                for ( int32_t i = 0; i < static_cast<int32_t>( _icnVsSprite[id].size() ); ++i ) {
```
Comment (ihhub) [inline src/fheroes2/agg/agg_image.cpp:827]: Could you please explain why we can't use a standard generated button?

**Review by ihhub — APPROVED** [review]

**Comment by Branikolog** [comment]
Hello, @zenseii 
What a pleasure to review such engine-changings PRs. :)
Won't the button look better, if we shift text 1px to the left. I feel, like it's not centered, or am I wrong?
image
I think, dismiss, upgrade and large exit/okay buttons have the same shifting?
Or is this just my tired eyes lying to me? :)

**Comment by Branikolog** [comment]
Hi, @zenseii 
Oh, I actually was talking about the text. But I haven't noticed shifted button itself. :) I suppose, the button is alligned from the left window border, so being expanded, the button expanded to the right side only....

**Comment by Branikolog** [comment]
Hi, @zenseii 
Now buttons look properly aligned and texts seem to be placed in center of buttons. But many of expanded buttons have issues with background near the right edge. I think, we should fix it within current PR but later, after the 1.0 release.

**Comment by Branikolog** [comment]
@zenseii 
If we merge this PR we'd have some sort of regression, regarding shadows of buttons. I know, that's not so essential, but I got used to the thing, that we're always going to only improve something in game. )

**Comment by ihhub** [comment]
Hi @Branikolog , can you point which buttons have issues? I will take a look how to fix them.

**Comment by Branikolog** [comment]
Hello, @ihhub 

> Hi @Branikolog , can you point which buttons have issues? I will take a look how to fix them.

I'm afraid, I won't be able to do that next two days. :) If you're not in hurry, I'll try to find all buttons up to 2nd January.

**Comment by ihhub** [comment]
Hi @zenseii , could you please rebased this branch from **master** and use new standard window to solve the outstanding problems with buttons?

**Review by Branikolog — APPROVED** [review]

**Comment by ihhub** [comment]
@zenseii , big thanks for this improvement!

**Pull request merged**